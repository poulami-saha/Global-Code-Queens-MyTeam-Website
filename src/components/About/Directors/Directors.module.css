* {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
}

.container {
  background-color: var(--deep-jungle-blue);
  display: flex;
  position: relative;
  justify-content: center;
  overflow-y: hidden;
}

.container::before,
.container::after {
  content: "";
  position: absolute;
}

.container::before {
  width: 200px;
  height: 200px;
  top: 0px;
  left: 0px;
  transform: translate(-50%, -50%);
  background-image: url("../../../assets/bg-pattern-about-2-contact-1.svg");
}

.container::after {
  width: 147px;
  height: 100px;
  bottom: 0px;
  right: 0px;
  background-image: url("../../../assets/bg-pattern-home-4-about-3.svg");
}

.article {
  max-width: 94vw;
  margin: 5.5rem auto;
}

.title {
  color: var(--primary-white);
  text-align: center;
}

.profiles {
  margin-top: 3rem;
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  align-content: space-between;
  row-gap: 1.5rem;
  column-gap: 0.688rem;
}

@media (min-width: 768px) {
  .container::before {
    transform: translate(-50%, 0%);
  }

  .topImage {
    top: 0;
  }

  .article {
    margin: 6.25rem auto;
  }

  .profiles {
    row-gap: 3rem;
  }
}

@media (min-width: 960px) {
  .article {
    margin: 8.75rem auto;
  }

  .profiles {
    margin-top: 4rem;
    column-gap: 1.875rem;
  }
}
