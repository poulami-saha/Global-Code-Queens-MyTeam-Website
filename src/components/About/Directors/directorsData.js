import nikita from "../../../assets/avatar-nikita.jpg";
import aden from "../../../assets/avatar-aden.jpg";
import christian from "../../../assets/avatar-christian.jpg";
import drake from "../../../assets/avatar-drake.jpg";
import cruz from "../../../assets/avatar-cruz.jpg";
import griffin from "../../../assets/avatar-griffin.jpg";

export const directors = [
  {
    name: "Nikita Marks",
    title: "Founder & CEO",
    quote:
      "Our mission is to empower talent to achieve their aspirations with integrity and passion.",
    avatar: nikita,
  },
  {
    name: "Cristian Duncan",
    title: "Co-founder & COO",
    quote:
      "Our success stems from teamwork, innovation, and a focus on delivering excellence every single day.",
    avatar: christian,
  },
  {
    name: "Cruz Hamer",
    title: "Co-founder & CTO",
    quote:
      "We are committed to to ensuring that our talent and clients have the best tools and platforms to succeed.",
    avatar: cruz,
  },
  {
    name: "Drake Heaton",
    title: "Business Development Lead",
    quote:
      "By cultivating strategic alliances, we aim to expand our footprint and amplify our industry impact.",
    avatar: drake,
  },
  {
    name: "Griffin Wise",
    title: "Lead Marketing",
    quote:
      "Unique perspectives shape unique products, which is what you need to survive these days.",
    avatar: griffin,
  },
  {
    name: "Aden Allan",
    title: "Head of Talent",
    quote:
      "Empowered teams create truly amazing products. Set the north star and let them follow it.",
    avatar: aden,
  },
];
