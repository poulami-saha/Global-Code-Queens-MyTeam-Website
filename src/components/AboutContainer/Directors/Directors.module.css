.container {
  background-color: var(--deep-jungle-blue);
  display: flex;
  position: relative;
  justify-content: center;
  overflow-y: hidden;
  background-image: url("../../../assets/bg-pattern-about-2-contact-1.svg"),
    url("../../../assets/bg-pattern-home-4-about-3.svg");
  background-position: top -100px left -100px,
    bottom right;
  background-repeat: no-repeat;
}

.article {
  width: 100%;
  margin: 5.5rem auto;
}

.title {
  color: var(--primary-white);
  text-align: center;
}

.profiles {
  max-width: 100%;
  margin-top: 3rem;
  display: grid;
  grid-template-columns: 1fr;
  row-gap: 3.5rem;
  justify-content: center;
}

@media (min-width: 768px) {
  .container {
    background-position: top left -100px,
      bottom right;
  }

  .topImage {
    top: 0;
  }

  .article {
    margin: 6.25rem auto;
  }

  .profiles {
    grid-template-columns: repeat(2, 1fr);
    row-gap: 3rem;
    column-gap: 0.688rem;
  }
}

@media (min-width: 960px) {
  .article {
    margin: 8.75rem 0;
  }

  .profiles {
    margin-top: 4rem;
    grid-template-columns: repeat(3, 1fr);
    column-gap: 2.0833%;
  }
}
