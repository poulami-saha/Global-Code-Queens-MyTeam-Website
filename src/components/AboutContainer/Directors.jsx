import { Typography } from "../utility";
import styles from "./Directors.module.css";
import ProfileCard from "./ProfileCard";

import { directors } from "./directorsData";

const Directors = () => {
  return (
    <section className={styles.container}>
      <article className={styles.article}>
        <Typography elType="h2" className={styles.title}>
          Meet the directors
        </Typography>
        <div className={styles.profiles}>
          {directors.map((director, idx) => (
            <ProfileCard key={idx} person={director} />
          ))}
        </div>
      </article>
    </section>
  );
};

export default Directors;
