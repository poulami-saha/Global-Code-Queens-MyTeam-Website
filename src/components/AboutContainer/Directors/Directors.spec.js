import { render, fireEvent, screen, within } from "@testing-library/react";
import Directors from "./Directors";
import { directors } from "./directorsData";

test("renders Directors section correctly and matches snapshot", async () => {
  const { asFragment } = render(<Directors />);

  // Snapshot test
  expect(asFragment()).toMatchSnapshot();

  // Content test
  expect(screen.getByText(/Meet the directors/)).toBeInTheDocument();
});

test("renders each director's name and title correctly", () => {
  render(<Directors />);

  directors.forEach((director) => {
    const { name, title } = director;

    const profileCard = screen.getByTestId(`profile-card-${name}`);
    const { getByText } = within(profileCard);

    // Content test
    expect(getByText(name)).toBeInTheDocument();
    expect(getByText(title)).toBeInTheDocument();
  });
});

test("renders each director's name and quote when button is clicked", () => {
  render(<Directors />);

  directors.forEach((director) => {
    const { name, quote } = director;

    const profileCard = screen.getByTestId(`profile-card-${name}`);

    const { getByRole, getByText } = within(profileCard);

    const button = getByRole("button");
    fireEvent.click(button);

    // Content test
    expect(getByText(name)).toBeInTheDocument();
    expect(getByText(`"${quote}"`)).toBeInTheDocument();
  });
});
