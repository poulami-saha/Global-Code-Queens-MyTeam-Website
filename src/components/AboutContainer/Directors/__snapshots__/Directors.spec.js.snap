// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Directors section correctly and matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="container"
  >
    <article
      class="article"
    >
      <h2
        class="title h2"
        id=""
      >
        Meet the directors
      </h2>
      <div
        class="profiles"
      >
        <article
          class="card"
          data-testid="profile-card-Nikita Marks"
        >
          <div
            class="avatarContainer"
          >
            <img
              alt="Nikita Marks"
              class="avatar"
              src="avatar-nikita.jpg"
            />
            <h3
              class="name h3 h3Large"
              id=""
            >
              Nikita Marks
            </h3>
            <p
              class="title body2"
              id=""
            >
              Founder & CEO
            </p>
          </div>
          <button
            class="button mainButton showButton buttonShowMore"
          >
            <svg>
              icon-cross.svg
            </svg>
          </button>
        </article>
        <article
          class="card"
          data-testid="profile-card-Cristian Duncan"
        >
          <div
            class="avatarContainer"
          >
            <img
              alt="Cristian Duncan"
              class="avatar"
              src="avatar-christian.jpg"
            />
            <h3
              class="name h3 h3Large"
              id=""
            >
              Cristian Duncan
            </h3>
            <p
              class="title body2"
              id=""
            >
              Co-founder & COO
            </p>
          </div>
          <button
            class="button mainButton showButton buttonShowMore"
          >
            <svg>
              icon-cross.svg
            </svg>
          </button>
        </article>
        <article
          class="card"
          data-testid="profile-card-Cruz Hamer"
        >
          <div
            class="avatarContainer"
          >
            <img
              alt="Cruz Hamer"
              class="avatar"
              src="avatar-cruz.jpg"
            />
            <h3
              class="name h3 h3Large"
              id=""
            >
              Cruz Hamer
            </h3>
            <p
              class="title body2"
              id=""
            >
              Co-founder & CTO
            </p>
          </div>
          <button
            class="button mainButton showButton buttonShowMore"
          >
            <svg>
              icon-cross.svg
            </svg>
          </button>
        </article>
        <article
          class="card"
          data-testid="profile-card-Drake Heaton"
        >
          <div
            class="avatarContainer"
          >
            <img
              alt="Drake Heaton"
              class="avatar"
              src="avatar-drake.jpg"
            />
            <h3
              class="name h3 h3Large"
              id=""
            >
              Drake Heaton
            </h3>
            <p
              class="title body2"
              id=""
            >
              Business Development Lead
            </p>
          </div>
          <button
            class="button mainButton showButton buttonShowMore"
          >
            <svg>
              icon-cross.svg
            </svg>
          </button>
        </article>
        <article
          class="card"
          data-testid="profile-card-Griffin Wise"
        >
          <div
            class="avatarContainer"
          >
            <img
              alt="Griffin Wise"
              class="avatar"
              src="avatar-griffin.jpg"
            />
            <h3
              class="name h3 h3Large"
              id=""
            >
              Griffin Wise
            </h3>
            <p
              class="title body2"
              id=""
            >
              Lead Marketing
            </p>
          </div>
          <button
            class="button mainButton showButton buttonShowMore"
          >
            <svg>
              icon-cross.svg
            </svg>
          </button>
        </article>
        <article
          class="card"
          data-testid="profile-card-Aden Allan"
        >
          <div
            class="avatarContainer"
          >
            <img
              alt="Aden Allan"
              class="avatar"
              src="avatar-aden.jpg"
            />
            <h3
              class="name h3 h3Large"
              id=""
            >
              Aden Allan
            </h3>
            <p
              class="title body2"
              id=""
            >
              Head of Talent
            </p>
          </div>
          <button
            class="button mainButton showButton buttonShowMore"
          >
            <svg>
              icon-cross.svg
            </svg>
          </button>
        </article>
      </div>
    </article>
  </section>
</DocumentFragment>
`;
